haystack/pipeline.py

```diff
@@ -136,6 +141,100 @@ def draw(self, path: Path = Path("pipeline.png")):
         graphviz.layout("dot")
         graphviz.draw(path)
 
+    @classmethod
+    def load_from_yaml(cls, path: Path, pipeline_name: Optional[str] = None):
+        """
+        Load Pipeline from a YAML file defining the individual components and how they're tied together to form
+        a Pipeline. A single YAML can declare multiple Pipelines, in which case an explicit `pipeline_name` must
+        be passed.
+
+        :param path: path of the YAML file.
+        :param pipeline_name: if the YAML contains multiple pipelines, the pipeline_name to load must be set.
+        """
+        with open(path, "r") as stream:
+            data = yaml.safe_load(stream)
+
+        if pipeline_name is None:
+            if len(data["pipelines"]) == 1:
+                pipeline_config = data["pipelines"][0]
+            else:
+                raise Exception("The YAML contains multiple pipelines. Please specify the pipeline name to load.")
+        else:
+            pipelines_in_yaml = list(filter(lambda p: p["name"] == pipeline_name, data["pipelines"]))
+            if not pipelines_in_yaml:
+                raise Exception(f"Cannot find any pipeline with name '{pipeline_name}' declared in the YAML file.")
+
+        definitions = {}  # definitions of each component from the YAML.
+        for definition in data["components"]:
+            cls._interpolate_environment_variables(definition)
```

Comment:
Let's make this optional by adding a boolean param to `load_from_yaml()`. As of now, this is quite hidden "auto magic". 
I would also vote for rather naming it `overwrite_with_env_variables`. Interpolating is a correct term here - but I personally associate it a lot with interpolation in mathematical terms and it confused me a bit in the beginning.

Reply:
We should also clarifyit in the Docstring of load_from_yaml()